### ISSUE TITLE
Question: Is there a roadmap for JavaFX introduction?

### ISSUE THREAD

**Issue by Oliver-Loeffler** [mediathekview/MediathekView#430]
Hi,

parts of MediathekView consist of JavaFX, others of Swing. Main GUI is Swing currently, mixing with JavaFX. 

Questions:
* Is there a readmap, a migration plan for JavaFX, which views, dialogs shall be migrated next?
* Is there the intention to fully move to JavaFX or stay with mixing Swing/JFX code?
* When going with JavaFX, is using FXML a welcomed approach (like using the *.form files)?

Thanks!

**Comment by derreisende77** [comment]
Hi,

yes I do have plans to fully migrate the current Swing UI to JavaFX, however it is unfortunately due to the age of the program not too easily to completely migrate everything to JavaFX quickly. 
On of the biggest problems I currently face is the support of three different OS (macOS, win, linux) with the application. Swing was quite easy but is dead. So I planned to migrate from Swing to JavaFX first via Interop, but this is a pretty dangerous thing and not too easy. 
JavaFX sounded like a good opportunity to revamp the UI BUT there are so many different problems on the various platforms that it really is a pain in the ass to write code with it. For example, JavaFX 11 crashes on HiDPI macOS Mojave, this is fixed with JavaFX 12. JavaFX 12 has UI problems with dialogs on KDE, fixed in JavaFX 13(early access). JavaFX Text does not render bold fonts correctly on macOS, I reported this several months ago and it still didn´t get fixed.
IMHO the whole java /javafx thing is still a big mess. Bug reporting is almost impossible and the JavaFX graphics system is unreliable.

Yes, if I plan to rewrite the client completely in JavaFX, it would use FXML. But currently I am more attracted to write native UIs for windows and macOS and leave linux support to someone who is using it ( I do all the coding on macOS and use VMs for the rest). 
And there is currently a big regression in TableView which will freeze the application after selecting all items if the item list is bigger than 5k items. I have to deal with 300k items....

So to summarize: Yes we have a migration plan, yes we would love to use JavaFX in the future and FXML, but it will be slow and hopefully they are fixing all existing problems by then. But at least for macOS I am evaluating another UI Kit for the client. For macOS the main reasons for not using java anymore are the new security features introduced with Mojave which prevent some features of MediathekView from working correctly on macOS. I don´t see a bright future for desktop java on macOS.

**Comment by derreisende77** [comment]
If you feel the mood to convert things to JavaFX/FXML I would be happy to try the PR

**Comment by Oliver-Loeffler** [comment]
Thanks, so I forked MediathekView and currently I am looking into the code to understand where what is and what each part is doing. Well, when I am ready, I will issue a PR.

**Comment by derreisende77** [comment]
btw are you familiar with FXML and Mig Layout class? I could never find useful docs on it and my FXML is too bad.

**Comment by Oliver-Loeffler** [comment]
No experience with Mig Layout but with FXML.

**Comment by derreisende77** [comment]
It is only bugfixes for the majority of bugs that paying customers experienced (Oracle side). No enhancements/deprecations or new controls like a macOS style popover (yes I know there is controlsfx - with its own problems). I haven't seen many fixes from Gluon and I am already on FX 19ea whatever.

TableView is still slow as f*** and JavaFX on macOS is still not able to render bold fonts OOTB.

There have been more and better contributions to the swing side (e.g. from Jetbrains) IMHO but I am more and more convinced that UI and Java are not a good fit anymore and maybe something like Qt would be more appropriate.
